### PR TITLE
Make sure all Matomo emails use correct branding.

### LINKED ISSUE #13251: Make all emails use the pretty HTML emails with branding
After #13250 theme variables will be defined in a class and the LESS variables will be generated in an event. Before these APIs are made public the following things need to be determined/done:

* Non-report emails need to be themed this way.
* Every part of report emails need to use the correct theme variables.
* If there are redundant properties in `EmailStyles` and `ThemeStyles` they should be resolved.
* If it should be allowed, PDF reports should be themed the same way as well.

At the moment the styling of reports/emails isn't entirely consistent so can't be made public API.

### PR DESCRIPTION
**Pull request by diosmosis**
Fixes #13251

### CODE REVIEW

File: core/View/HtmlReportEmailHeaderView.php [hunk 1]
```diff
@@ -77,7 +78,13 @@ public static function assignCommonParameters(View $view)
         $view->isCustomLogo = $customLogo->isEnabled() && CustomLogo::hasUserLogo();
         $view->logoHeader = $customLogo->getHeaderLogoUrl($pathOnly = false);
 
-        $view->hasWhiteLabel = \Piwik\Plugin\Manager::getInstance()->isPluginLoaded('WhiteLabel');
+        $view->hasWhiteLabel = \Piwik\Plugin\Manager::getInstance()->isPluginLoaded('WhiteLabel')
+            && \Piwik\Plugin\Manager::getInstance()->isPluginActivated('WhiteLabel')
+            && \Piwik\Plugin\Manager::getInstance()->isPluginInFilesystem('WhiteLabel');
+
+        $view->idSite = Common::getRequestVar('idSite', false);
```
Comment (tsteur) [inline core/View/HtmlReportEmailHeaderView.php]: Not sure about those 3 lines. Are they needed @diosmosis ? For what? Cause the current set `idSite` in the request might not be related to any sent email. Same for period or date. Sometimes  there eg hooks that listen to something like a plugin update, and then send an email about something totally else.

**Comment by tsteur** [comment]
Not sure if it is due to this PR but the sender is "Matomo Reports"... even when eg getting update notification 
image

**Comment by tsteur** [comment]
Likely unrelated to this... noticed the bullet points are smaller in the scheduled reports than the rest of the font... didn't look in the code if it was before already like this

image

**Comment by tsteur** [comment]
I suppose Custom alerts will need to be updated as well? Possibly also the mail send in QueuedTracking https://github.com/matomo-org/plugin-QueuedTracking/blob/3.3.1/Tasks.php#L73-L84 but not too important

File: core/View/HtmlReportEmailHeaderView.php (hunk 1 above)
Comment (diosmosis) [reply core/View/HtmlReportEmailHeaderView.php]: These three params are used in the dashboard link in the email header. Some emails had it, some didn't. Thought I'd add it to all if the params were available.

Comment (tsteur) [reply core/View/HtmlReportEmailHeaderView.php]: Wouldn't we want to link the the default report?

Comment (diosmosis) [reply core/View/HtmlReportEmailHeaderView.php]: This is how the code worked before, but I can change it to link to what's configured in settings.

Comment (diosmosis) [reply core/View/HtmlReportEmailHeaderView.php]: Nvm, it already does this (by linking to CoreHome.index).

**Comment by diosmosis** [comment]
> Not sure if it is due to this PR but the sender is "Matomo Reports"... even when eg getting update notification

Looks like that's the default report name if `[General] noreply_email_name` isn't set. Wonder if it should just be from `Matomo` or something?

File: core/View/HtmlReportEmailHeaderView.php (hunk 1 above)
Comment (tsteur) [reply core/View/HtmlReportEmailHeaderView.php]: Wouldn't it need to link to `CoreHome.redirectToCoreHomeIndex` (requires no site, period, date, ... AFAIK)? I guess the only time we wouldn't want this is when a report is sent for a specific site. In all cases might make sense to load the default report and default time to load.

Comment (diosmosis) [reply core/View/HtmlReportEmailHeaderView.php]: Yes, would need to link to `redirectToCoreHomeIndex` or I guess to omit the action. I can make this change, except when a specific site is set.

**Comment by diosmosis** [comment]
@tsteur re the bullet points, does this look better?

image

**Comment by diosmosis** [comment]
Forgot to create the PR for CustomAlerts, can find it here: https://github.com/matomo-org/plugin-CustomAlerts/pull/54

**Comment by diosmosis** [comment]
Created a queued tracking PR as well: https://github.com/matomo-org/plugin-QueuedTracking/pull/93

**Comment by tsteur** [comment]
@diosmosis it should be fine as long as it is using the same font size 👍

**Comment by diosmosis** [comment]
It is now, think it was using the default chrome font size before.

**Comment by tsteur** [comment]
@diosmosis had a look at the other PRs and looks all good. I reckon can be merged as soon as tests succeed 👍

**Pull request merged**